app/battle.py

```diff
@@ -0,0 +1,16 @@
+from typing import Dict
+
+
+def battle(knight1: Dict, knight2: Dict) -> Dict[str, int]:
+    knight1["hp"] -= knight2["power"] - knight1["protection"]
+    knight2["hp"] -= knight1["power"] - knight2["protection"]
```

Comment:
Similarly, ensure that `knight1['power'] - knight2['protection']` does not result in a negative value by using `max(0, knight1['power'] - knight2['protection'])`.